pkg/trace/info/info.go

```diff
@@ -257,11 +258,16 @@ func InitInfo(conf *config.AgentConfig) error {
 			return
 		}
 
+		scrubbed, err := scrubber.ScrubBytes(buf)
+		if err != nil {
+			return
+		}
```

Comment:
This is going to cause a nil error to return since you're using `:=` instead of `=` here. See https://go.dev/play/p/K7sxXE2xvLp to demonstrate.
This part of the code feels a bit brittle in that regard though, so might be worth a refactor just to make sure that kind of thing can't happen.

Reply:
Ideally we could also unit test this to verify.

Reply:
Really nice catch! Thank you!
To make the code more robust, I moved the init logic to a separate function that returns an error ([commit](https://github.com/DataDog/datadog-agent/pull/15591/commits/e8cd28cc7bc760c12e8b586bda981d82b9cdc3b0)).
This way we keep the error declaration and its assignment very close in the code. Also added a comment with a link to the Go playground. Let me know if you think of a better approach!

Reply:
That looks good, thanks!!